replicate/common/generic-replicator.go

```diff
@@ -434,6 +474,25 @@ func (r *GenericReplicator) ResourceDeletedReplicateTo(source interface{}) {
 			r.DeleteResources(source, list, filters)
 		}
 	}
+
+	// delete replicated resources in namespaces that match labels
+	namespaceSelectorString, replicateToMatching := objMeta.GetAnnotations()[ReplicateToMatching]
+	if replicateToMatching {
+		namespaceSelector, err := labels.Parse(namespaceSelectorString)
+		if err != nil {
+			err = errors.Wrapf(err, "Failed parse namespace selector: %v", err)
+			logger.WithError(err).Errorf("Could not get namespaces: %+v", err)
+		}
```

Comment:
Currently, this function will continue, even on an error while parsing the namespace selector (which is a likely occurrence, since it's user input).

I'm not entirely sure what's going to happen in this case. The worst case imaginable is that `namespaceSelector` will be an empty selector that matches _all_ namespaces, causing a Resource to be replicated _everywhere_ against the users intent.